src/span.js

```diff
@@ -144,6 +144,17 @@ export default class Span {
      **/
     setOperationName(operationName: string): Span {
         this._operationName = operationName;
+        // If the span has debug set, then the question of resampling
+        // after setting the operation name is irrelevant.  The span is always
+        // sampled.
+        if (!this._spanContext.isDebug()) {
+            let sampler = this.tracer()._sampler;
+            var tags = {};  // TODO:oibe shouldn't this be initialized if not passed?
+            if (sampler.isSampled(operationName, tags)) {
```

Comment:
Here's a scenario for unit test - initial sampling set the tags based on default sampler, and re-sampling produces different sampling tags from a different type of sampler.